vendor/sigs.k8s.io/mcs-api/pkg/apis/v1alpha1/serviceexport.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+
+// ServiceExport declares that the Service with the same name and namespace
+// as this export should be consumable from other clusters.
+type ServiceExport struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// status describes the current state of an exported service.
+	// Service configuration comes from the Service that had the same
+	// name and namespace as this ServiceExport.
+	// Populated by the multi-cluster service implementation's controller.
+	// +optional
+	Status ServiceExportStatus `json:"status,omitempty"`
+}
+
+// ServiceExportStatus contains the current status of an export.
+type ServiceExportStatus struct {
+	// +optional
+	// +patchStrategy=merge
+	// +patchMergeKey=type
+	// +listType=map
+	// +listMapKey=type
+	Conditions []ServiceExportCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+// ServiceExportConditionType identifies a specific condition.
+type ServiceExportConditionType string
```

Comment:
Q: in other parts of our API the condition types are also strings, over time we added more and more conditions as strings, and now it's hard to keep track of all our possible conditions :-(

does it make sense to make this type more limiting, so we will have to update it each time we add new condition type, making a source of truth for all our possible condition for this CR ?

cc:// @solenoci hi, if we made the Plan condition types more strict, would it help with your task of creating a unified "state" field ?

Reply:
> Forklift is not the only possible writer,

Ahh, i missed that, why would some other operator change the status of a resource Forklift is managing ?

Reply:
No reason some other operator would change it, but because CRs _can_ be changed by an API user (even though it's unlikely) I think it's important to consider that when considering how to restrict field values (to make sure we either completely prevent values we don't expect via something like CRD validation or are tolerant of unexpected values).

Reply:
Sorry for taking over this PR, this discussion is a bit off topic :-)

> Having less conditions 

My concern is about havin "known" conditions, for example if you want to build a method that will take the conditions and return one "status" code, it is helpful that you covered all the checked all the used conditions types. it the condition type is a string, you never know if you tested for all available conditions.